### PR TITLE
[Intl] Removed deprecated code

### PR DESCRIPTION
**Pull request by malteschlueter**
| Q             | A
| ------------- | ---
| Branch?       | 6.0
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

This remove deprecated code from Intl.

TODO
- [x] Update UPGRADE-5.3.md and UPGRADE-6.0.md in Branch 5.3 #41362

### CODE REVIEW

**Comment by malteschlueter** [comment]
What should I do within the `Resources` directory? I'm not sure. Do I have to delete there something?

File: src/Symfony/Component/Intl/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,11 @@
 CHANGELOG
 =========
 
+6.0
+---
+ * Remove `DateFormatter` namespace, use symfony/polyfill-intl-icu ^1.21 instead
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Intl/CHANGELOG.md]: missing blank line

**Review by nicolas-grekas — COMMENTED** [review]
rebase needed

**Comment by nicolas-grekas** [comment]
Thank you @malteschlueter.

**Pull request merged**